include/glow/Support/Error.h

```diff
@@ -98,22 +98,41 @@ class GlowErr final : public llvm::ErrorInfo<GlowErr> {
     }
   }
 
-  GlowErr(llvm::StringRef fileName, size_t lineNumber, llvm::StringRef message,
-          ErrorCode ec)
-      : lineNumber_(lineNumber), fileName_(fileName), message_(message),
-        ec_(ec) {}
-
+/// Construct a GlowErr for an error at a given \p lineNumber in a given \p
+/// fileName with a given ErrorCode \p ec. The final arguments to this
+/// constructor are a printf-style format string and the accompanying printf
+/// arguments from which the error message will be constructed.
+/// NOTE: This disables format-security warnings that enforce only string
+/// literals be passed to strFormat but GlowErr should only be constructed with
+/// formatStrings as string literals.
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wformat-security"
```

Comment:
This is a pretty big wtf, if anyone has any suggestions for improving it, I'm interested in hearing

Reply:
Yeah it's a weird one, I can only assume that one of the inputs is not const in some invocation. Can't see it in this diff though.

Do you still see the error if you use `const char *messageFormat` rather than a const ref to an array?